add: реворк свармеров [IDB IGNORE] [MDB IGNORE]

NuSanya opened the pull request:
## Что этот ПР делает

1) Реворк свармеров, следуя следующему ТЗ: https://docs.google.com/document/d/1TeLjwrMbjXY1RUgZCUXlE8QySlEscRVh/edit
2) Добавляет стрельбу залпом турелям, дополняет стрельбу залпом мобам, добавляя рабочий разброс.
3) Добавляет новые статус эффекты на блокировку лечения стамины, и на блокировку метаболизации реагентов.
4) Убирает маяк Свармеров с Лаваленда и связанные с ним достижения.
5) Слегка меняет схему таргетинга у турелей, теперь по лежачим целям нелетальные турели будут продолжать стрелять.

Некоторое время я буду сам, для лучшего понимания того, как этот будет играться, спавнить свармеров. Вес у ивента будет на это время ноль.

## Почему это хорошо для игры

Реворк мёртвого антагониста на командную игру с более глубокими механиками.

## Демонстрация изменений
<details><summary>Изображения и видео</summary>
<p>
<img width="493" height="474" alt="image" src="https://github.com/user-attachments/assets/b8ec8666-e5e1-40b6-b8fc-5ee43b1a4649" />
</p>
</details>

## Список изменений
:cl:
add: Реворк свармеров
/:cl:

NuSanya committed:
initial

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
linters

NuSanya committed:
mega swarmer + tweaks

NuSanya committed:
add core move action + bugfixes

NuSanya committed:
comment achievement

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
upstream projectiles dmi file move

NuSanya committed:
add objective, mega swarmer announce, todo cleanup

NuSanya committed:
swarmer event

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
update tgui

NuSanya committed:
linter

PlayerUnknown14 commented:
/gemini review

NuSanya committed:
localization + fixes

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

PlayerUnknown14 commented:
/gemini review

NuSanya committed:
more fixes

NuSanya committed:
even more fixes

NuSanya committed:
more code improvs

NuSanya commented:
/gemini review

NuSanya committed:
add landmarks + event tweaks

NuSanya commented:
/gemini review

NuSanya commented on code/game/machinery/portable_turret.dm:
```diff
@@ -1117,6 +1121,195 @@ GLOBAL_LIST_EMPTY(turret_icons)
+	return 10 // Swarmer turrets shoot everything not in their faction
+
+/obj/machinery/porta_turret/swarmer/emp_act(severity)
+	..()
+	take_damage(SWARMER_EMP_DAMAGE)
```
сигнал в проке атома

NuSanya commented on code/modules/antagonists/swarmers/swarmer_structures.dm:
```diff
@@ -0,0 +1,869 @@
+	var/list/potential_hubs = list()
+	var/list/hub_names = list()
+	var/list/duplicate_hub_count = list()
+	for(var/obj/structure/swarmer/transport_hub/hub in GLOB.swarmer_objects)
```
нет

NuSanya committed:
fixes + core destroying optimization

NuSanya committed:
forgor fast_load define

NuSanya committed:
fixes

NuSanya commented:
/gemini review

NuSanya commented on code/game/machinery/portable_turret.dm:
```diff
@@ -732,29 +733,32 @@ GLOBAL_LIST_EMPTY(turret_icons)
+		playsound(loc, shot_sound, 75, TRUE)
+	if(!proj)
+		return
+	if(!istype(proj))
+		proj.throw_at(target, scan_range, 1)
+		return
```
proj от obj/projectile, проверка для турелей где снаряд зачем-то не пуля

NuSanya committed:
remove debug parameters

NuSanya committed:
Merge branch 'master220' into idk-swarmer

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
tgui

PlayerUnknown14 commented:
Позже осмотрю.

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
merge conflict stuff

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

AurothWyvern commented on code/modules/antagonists/swarmers/action.dm:
```diff
@@ -0,0 +1,242 @@
+
+/datum/action/cooldown/swarmer/build/processer
+	name = "Создать переработчик органики"
+	desc = "Обрабатывает неживую материю."
```
Как будто по описанию должно быть либо **В** неживую перерабатывает, либо перерабатывает органическую

AurothWyvern commented on code/modules/antagonists/swarmers/action.dm:
```diff
@@ -0,0 +1,242 @@
+
+/datum/action/cooldown/swarmer/build/acp_turret
+	name = "Создать установку ACP"
+	desc = "Турель, бьющая целей по области, накладывая дебаффы."
```
возможно стоит уточнить какие дебаффы

AurothWyvern commented on code/modules/antagonists/swarmers/swarmer.dm:
```diff
@@ -0,0 +1,512 @@
+	target.mind.transfer_to(new_swarmer)
+	balloon_alert(src, "успех!")
+	new_swarmer.spark_system.start()
+	add_conversion_logs(target, "Converted into [new_swarmer.name].")
```
возможно не переведено

AurothWyvern commented on code/modules/antagonists/swarmers/swarmer_structures.dm:
```diff
@@ -0,0 +1,878 @@
+ */
+/obj/structure/swarmer/organic_processer
+	name = "swarmer organic processer"
+	desc = "Переработчик, позволяющий обрабатывать органику в ресурсы \"Свармеров\"."
```
скорее "перерабатывать"

AurothWyvern commented on code/modules/antagonists/swarmers/swarmer_structures.dm:
```diff
@@ -0,0 +1,878 @@
+/obj/structure/swarmer/organic_processer
+	name = "swarmer organic processer"
+	desc = "Переработчик, позволяющий обрабатывать органику в ресурсы \"Свармеров\"."
+	swarmer_examine = "Обрабатывает овощи, реагенты. Не делает этого в открученном состоянии. Загрузка в эту машину происходит через атаку по данным предметам."
```
тут тоже скорее "перерабатывать"

AurothWyvern reviewed (commented):
Оставил набор небольших комментариев перепроверить автору. В остальном вроде везде всё без опечаток и ошибок в ру тексте.

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
merge conflict changes

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
merge conflict markers

NuSanya committed:
sibyl update

NuSanya committed:
more linter

NuSanya committed:
wow more linter

NuSanya committed:
wow skobochka

NuSanya committed:
un-fuck maps

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
tgui

NuSanya committed:
redo swarmer_act

NuSanya committed:
code improvements

swarmer act tweaks + combat swarmer action redo to element + some sound effects

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
tgui

NuSanya committed:
remove some istype checks for easier review

NuSanya committed:
mistypo

NuSanya committed:
linter

NuSanya committed:
i may not know how to do linters locally

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
upstream tweaks

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
structurize, sounds + animation tweaks

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
reduce event weight to 0, tweak balance

NuSanya committed:
add spawners to maps

NuSanya committed:
balance tweaks

NuSanya committed:
event tweaks

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
spacing

NuSanya committed:
Update tgui.bundle.js

NuSanya committed:
add swarmer catwalks, add cd for every action

NuSanya committed:
slight balance tweaks

NuSanya committed:
makes analyzer's occupant safe from pressure

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
tgui

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
Merge remote-tracking branch 'upstream/master220' into idk-swarmer

NuSanya committed:
Revert "Merge remote-tracking branch 'upstream/master220' into idk-swarmer"

This reverts commit 2019707927039aa9d079afa4fa13e864e49995ce, reversing
changes made to a69140af316d649693658d59d79dd8f59c179189.

NuSanya committed:
Revert "Merge remote-tracking branch 'upstream/master220' into idk-swarmer"

This reverts commit a69140af316d649693658d59d79dd8f59c179189, reversing
changes made to c2d2e963136189a0f4bf0b5ba4bf709ff90cd0d3.

NuSanya committed:
Reapply "Merge remote-tracking branch 'upstream/master220' into idk-swarmer"

This reverts commit 2e7b368ee4ad9cf60fc6a31b683454168207abe9.

NuSanya committed:
Reapply "Merge remote-tracking branch 'upstream/master220' into idk-swarmer"

This reverts commit 086cfe729f2c7aa7f554d4b1194c239e1463c2a7.

NuSanya committed:
upstream fixes

NuSanya committed:
tgui
